cpp-client/deephaven/dhclient/src/utility/table_maker.cc

```diff
@@ -59,11 +67,64 @@ TableHandle TableMaker::MakeTable(const TableHandleManager &manager) {
   return manager.MakeTableHandleFromTicket(std::move(ticket));
 }
 
+std::shared_ptr<arrow::Table> TableMaker::MakeArrowTable() const {
+  auto schema = MakeSchema();
+
+  // Extract the data_vec column.
+  auto data_vec = MakeReservedVector<std::shared_ptr<arrow::Array>>(column_infos_.size());
+  for (const auto &info : column_infos_) {
+    data_vec.push_back(info.data_);
+  }
+  auto data = GetColumnsNotEmpty();
+  return arrow::Table::Make(std::move(schema), std::move(data));
+}
+
+std::vector<std::shared_ptr<arrow::Array>> TableMaker::GetColumnsNotEmpty() const {
+  std::vector<std::shared_ptr<arrow::Array>> result;
+  for (const auto &info : column_infos_) {
+    result.emplace_back(info.data_);
+  }
+  if (result.empty()) {
+    throw std::runtime_error(DEEPHAVEN_LOCATION_STR("Can't make table with no columns"));
+  }
+  return result;
+}
+
+std::shared_ptr<arrow::Schema> TableMaker::MakeSchema() const {
+  arrow::SchemaBuilder sb;
+  for (const auto &info : column_infos_) {
+    auto kv_metadata = std::make_shared<arrow::KeyValueMetadata>();
+    OkOrThrow(DEEPHAVEN_LOCATION_EXPR(kv_metadata->Set("deephaven:type",
```

Comment:
Is `"deephaven:type"` something that would make sense to make a constant somewhere?

Reply:
Done, PTAL